### PR TITLE
Remove Git Branching Requirement

### PR DESCRIPTION
**Pull request by abhineet97**
## Description

Vincent Driessen's Git Branching model (also known as [git-flow](https://nvie.com/posts/a-successful-git-branching-model/)) was proposed more than 10 years ago. It was meant for closed-source projects and wasn't really meant for open-source projects. Seeing that it is _required_ might confuse and deter new contributors.

Kindly consider removing this requirement.

## Type of change

- [X] This change requires a documentation update

### CODE REVIEW

**Comment by gmajan** [comment]
Hi Abhineet,
Thanks for the suggestion. I am open to change it. Can you let me know what we are replacing it with?
Can you also get the view  of @Tlazypanda and @Patamap 
Thx.

**Comment by abhineet97** [comment]
Hello @gmajan 

Thanks GitHub's pull request mechanism, there's no need for a branching model. This is because in a PR, it doesn't really matter what the **from** branch is. The only thing that's important for us is that the **to** branch is `master`.

A formal and interactive explanation of this way of using Git & GitHub is available [here](https://guides.github.com/introduction/flow/)

@Tlazypanda and @Patamap can you please weigh in on this and maybe share some insights of how other open source projects do this? Thanks!

**Comment by gmajan** [comment]
@abhineet97 Abhineet,
I do not agree with your view. The GitHub approach works for CI and for software running in 1 place. But our software is running as a particular version on various machines. So we need to "release" versions that will not change and can be found again in the future because a country will calculate emissions, then the results are submitted to the United Nations, then an independent verifier is appointed who will have to use the same version of FLINT to calculate the emissions that were reported to the UN. So we are stuck with master versions and development versions. We can be flexible about how we deal with features since we do not have too many that are developed in parallel (at this moment) and we can have those integrated continuously. (Once we set up CI.) 
Would you agree with me?
Thx

**Comment by abhineet97** [comment]
Thanks for the explanation @gmajan.

I understand that you have a concern that old versions of FLINT should be always easily accessible. I believe that this is already addressed by the use of GitHub's Releases feature [here](https://github.com/moja-global/FLINT/releases). Please let me know if this isn't what you were implying.

Now, the thing with a branching model like git-flow is that it's _impossible_ for people without repo access to follow. This is because GitHub doesn't allow contributors without repo access to submit PRs to branches that don't already exist. This means that a contributor can't even submit a simple bug fix if they don't have repo access. This is because GitHub won't allow them to create the `hotfix-1.0.1` branch which is required by git-flow.

I agree that having seperate branches for different in-development features would make a lot of sense and would allow moja great flexibility. This is something that moja should do on its end. I don't think contributors need to follow git-flow for this.

I hope that my explanations are making sense :sweat_smile:
Thanks!

**Comment by Patamap** [comment]
I prefer to have the current way of working. However, it would be good to hear the comment from the maintenance team  @leitchy @kaskou @malfrancis ?  Thanks

**Comment by gmajan** [comment]
Abhineet, I see your point. 
I have been thinking only about contributions to existing branches. Hmmm.
Ok. Let's fix this. 
So you are proposing GitHub Flow with features if we have them (which we are not really using anyway at the moment but might in the future). Is that correct? Pls confirm.

I am happy with that if the maintainers are. 

Once we have confirmation from the maintainers, we need to update the docs. Can you do that? thx.

Thank you for insisting.

**Comment by abhineet97** [comment]
>  So you are proposing GitHub Flow with features if we have them (which we are not really using anyway at the moment but might in the future). Is that correct? Pls confirm.

My suggestion is that we remove the line from our contribution guidelines that mandates that a contributor must follow Vincent Driessen's Git Branching model. I believe that it not only causes confusion but is also impossible for a contributor to adhere to.

I made this PR because I felt that this requirement might deter new contributions. I did not intend to propose any changes to the way in which moja and its maintainers work. I'm sorry if I wasn't clear before about that before.

Thank you so much for considering my view. I'm immensly grateful for your understanding and patience.

**Comment by gmajan** [comment]
@abhineet97 
I understand the change you are proposing in contributing.md 
It is great that you are making this proposal and I see that there is merit in your view.
I am just a bit reluctant to approve the pull request because:

1. Behind your contributing.md there is also additional documentation that we should change if we drop it from the contributing.md. So we need to delete the info in those places as well. 
2. Moreover, it is not enough to remove references to Git-Flow but we should indicate that the process we are following (even if we follow GitHub Flow, we should tell contributors, no?) 
3. Finally, we need to inform our maintainers and contributors (even if we do not change the way we work) to make sure that we all communicate the same approach. 

So basically, I think just removing 1 sentence will not resolve this issue. It is probably slightly bigger than this 1 sentence. I agree that we need to correct the information, but I would like to do it in a systematic manner. 

It would be great if you can do it. If not, could you pls submit an issue so somebody else can take this up. 
Thank you

**Comment by abhineet97** [comment]
@gmajan, thanks for clarifying.

Based on what I understood, here's what I think I'll do:

I'll update this PR after doing all the necessary changes to the docs (as described in 1 and 2 above) and then you can merge it after we've successfully communicated with the maintainers (as suggested in in 3 above).

Do you think this will be alright?

Thanks!

**Comment by gmajan** [comment]
That sounds wonderful. Thank you very much. I am looking forward to the changes. Thank you!!!

**Comment by abhineet97** [comment]
@gmajan, I've removed references to git-flow.

Can you please suggest me how I should inform contributors about GitHub flow. Should I create a new doc under `Contributing`?

**Comment by Patamap** [comment]
@abhineet97  It would be better that you seek the agreement of your proposal new way of working before writing the document and coding.  Just in case, the maintanent team may disagree with this proposal or change the proposal. @leitchy @malfrancis @mfellows, they are the maintenance team.

File: Contributing/How-to-Contribute-Code.md [hunk 1]
```diff
@@ -24,4 +21,4 @@
     * Ensure your Pull-Request passes all checks and has no conflicts
```
Comment (gmajan) [inline Contributing/How-to-Contribute-Code.md:21]: Should we have a bit more info here? 
I want to keep things as simple as possible but on the other hand we need to be complete in order to avoid confusion.  So do you want to mention branches here or not?
Thx

**Comment by gmajan** [comment]
I have reviewed your changes and they look fine. I have left 1 small comment in your text. thx. Nice work.

**Comment by Patamap** [comment]
@gmajan how do you plan to inform the team of this change?

File: Contributing/How-to-Contribute-Code.md (hunk 1 above)
Comment (abhineet97) [reply Contributing/How-to-Contribute-Code.md:21]: I agree that we should document our branching model. But I was not sure what to add since the maintainers have not weighed-in on this yet.
